tokio-macros/src/entry.rs

```diff
@@ -1,11 +1,171 @@
 use proc_macro::TokenStream;
+use proc_macro2::Span;
 use quote::quote;
-use std::num::NonZeroUsize;
+use syn::spanned::Spanned;
 
 #[derive(Clone, Copy, PartialEq)]
-enum Runtime {
-    Basic,
-    Threaded,
+enum RuntimeFlavor {
+    InPlace,
+    WorkStealing,
+}
+
+impl RuntimeFlavor {
+    fn from_str(s: &str) -> Result<RuntimeFlavor, String> {
+        match s {
+            "in_place" => Ok(RuntimeFlavor::InPlace),
+            "work_stealing" => Ok(RuntimeFlavor::WorkStealing),
+            "single_threaded" => Err(format!("The single threaded runtime flavor is called `in_place`.")),
+            "basic_scheduler" => Err(format!("The `basic_scheduler` runtime flavor has been renamed to `in_place`.")),
```

Comment:
do we need to do this? I'd rather not leave "old" stuff in a breaking change.

Reply:
No. I added it to ease the transition for users coming from v0.2, and because it wouldn't be a breaking change to remove later.

Reply:
Hmm this will just generate a compile failure right? Then I think that is fine.